QRHunter/app/src/main/java/com/example/qrhunter/ui/scan/AfterScanFragment.java

```diff
@@ -1,35 +1,194 @@
 package com.example.qrhunter.ui.scan;
 
+import android.Manifest;
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.location.Address;
+import android.location.Geocoder;
 import android.os.Bundle;
+import android.provider.MediaStore;
+import android.util.Base64;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
 
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
 import androidx.fragment.app.Fragment;
 import androidx.lifecycle.ViewModelProvider;
+import androidx.navigation.Navigation;
 
+import com.example.qrhunter.MainActivity;
+import com.example.qrhunter.R;
+import com.example.qrhunter.data.model.Comment;
+import com.example.qrhunter.data.model.Location;
+import com.example.qrhunter.data.model.QRCode;
+import com.example.qrhunter.data.repository.QRCodeRepository;
+import com.example.qrhunter.utils.QRCodeName;
+import com.example.qrhunter.utils.QRCodeUtil;
+import com.example.qrhunter.utils.QRCodeVisual;
 import com.example.qrhunter.databinding.FragmentAfterScanBinding;
+import com.google.firebase.firestore.FirebaseFirestore;
 
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * screen segment after scan
+ */
 public class AfterScanFragment extends Fragment {
     private FragmentAfterScanBinding binding;
+    private String hashValue;
+    private final int MY_PERMISSIONS_REQUEST_CAMERA = 1001;
+    ActivityResultLauncher<Intent> cameraActivityResultLauncher;
+
+    Bitmap savedPhoto = null;
 
+    // location is 0,0 until user decides to record location
+    private int latitude = 0;
+    private int longitude = 0;
+    boolean deletePhoto = false;
+    boolean canSaveValue = false;
+    //
```

Comment:
Ideally, these states (including hashValue) that are used for UI would go into the [ViewModel class](https://developer.android.com/topic/libraries/architecture/viewmodel). Optional though